apps/dapp-console-api/src/routes/wallets/WalletsRoute.ts

```diff
@@ -0,0 +1,43 @@
+import { isPrivyAuthed } from '@/middleware'
+import { metrics } from '@/monitoring/metrics'
+import { Trpc } from '@/Trpc'
+import { fetchAndSyncPrivyWallets } from '@/utils'
+
+import { Route } from '../Route'
+
+export class WalletsRoute extends Route {
+  public readonly name = 'wallets' as const
+
+  public readonly syncWallets = 'syncWallets' as const
+  public readonly syncWalletsController = this.trpc.procedure
+    .use(isPrivyAuthed(this.trpc))
+    .mutation(async ({ ctx }) => {
+      try {
+        const { user } = ctx.session
+
+        await fetchAndSyncPrivyWallets(
+          this.trpc.database,
+          this.trpc.privy,
+          user!.privyDid,
+          user!.entityId,
+        )
+
```

Comment:
one thing i think we'll want to add to this endpoint in a follow on is to loop thru all the newly added wallets and do the verification checks

Reply:
yeah, that is a good idea